File: src/Unidesk/Unidesk/Dtos/ThesisUserDto.cs
using System.ComponentModel.DataAnnotations;
using Unidesk.Db.Models;
using Unidesk.Server;

namespace Unidesk.Dtos;

public class ThesisUserDto
{
    [Required]
    public UserDto User { get; set; }
    [Required]
    public Guid UserId { get; set; }
    
    [IgnoreMapping]
    public ThesisDto Thesis { get; set; }
    [Required]
    public Guid ThesisId { get; set; }
    
    public UserFunction Function { get; set; }
}

File: src/Unidesk/Unidesk.Client/src/api-client/models/ThesisUserDto.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ThesisDto } from './ThesisDto';
import type { UserDto } from './UserDto';
import type { UserFunction } from './UserFunction';

export type ThesisUserDto = {
    user: UserDto;
    userId: string;
    thesis?: ThesisDto;
    thesisId: string;
    function?: UserFunction;
};
